cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -82,23 +88,81 @@ pub struct CheckedBuildMetadata {
     _private: (),
 }
 
+#[allow(async_fn_in_trait)]
+pub trait BinaryCache {
+    async fn upload(&self, path: &str) -> Result<(), PublishError>;
+    fn cache_uri(&self) -> &Url;
+}
+
+pub struct NixCopyCache {
+    pub uri: Url,
+    pub key_file: String,
+}
+
+impl BinaryCache for NixCopyCache {
+    async fn upload(&self, path: &str) -> Result<(), PublishError> {
+        debug!("Uploading {} to cache...", path);
+        let uri_with_key = format!("{}?secret-key={}", self.uri.as_str(), self.key_file);
+        let mut copy_command = Command::new(&*NIX_BIN);
+        copy_command
+            .arg("copy")
+            .arg("--to")
+            .arg(uri_with_key)
+            .arg(path);
```

Comment:
with the use of `nix` commands proliferating, we might want to centralize at least the nix config parts, i.e. the --store option settings and ensuring other necessary settings such that user configs have less impact.
fine to set this aside for a second, although the store part will be needed to support testing in nix sandboxes